### PR TITLE
Improve docs of convolution_2d

### PR DESCRIPTION
**Pull request by keisuke-umezawa**
This is PR for improving docs of functions/links. Related issue: #2182

Modified functions/links:
 - functions/connection/convolution_2d

Doctest result is here.
```
Document: reference/functions
-----------------------------
1 items passed all tests:
 106 tests in default
106 tests in 1 items.
106 passed and 0 failed.
Test passed.
```

### CODE REVIEW

File: chainer/functions/connection/convolution_2d.py [hunk 1]
```diff
@@ -308,58 +308,91 @@ def convolution_2d(x, W, b=None, stride=1, pad=0, use_cudnn=True,
...
-    :math:`(h_O, w_O)` is determined by the following equations:
+    Let :math:`(s_Y, s_X)` be the stride of filter application. Then, the
+    output size :math:`(h_O, w_O)` is determined by the following equations:
 
     .. math::
 
-       h_O &= (h + 2p_H - k_H) / s_Y + 1,\\\\
-       w_O &= (w + 2p_W - k_W) / s_X + 1.
+       h_O &= (h_I + 2h_P - h_K) / s_Y + 1,\\\\
+       w_O &= (w_I + 2w_P - w_K) / s_X + 1.
```
Comment (delta2323) [inline chainer/functions/connection/convolution_2d.py]: Could you write the formula when `cover_all` is `True`, too, as it changes the output size.

File: chainer/functions/connection/convolution_2d.py [hunk 2]
```diff
@@ -308,58 +308,91 @@ def convolution_2d(x, W, b=None, stride=1, pad=0, use_cudnn=True,
...
+        >>> b.shape
+        (1,)
+        >>> s_y, s_x = 2, 2
+        >>> y = F.convolution_2d(x, W, b, stride=(s_y, s_x), pad=(h_p, w_p))
+        >>> y.shape
+        (10, 1, 16, 21)
+        >>> h_o = int((h_i + 2 * h_p - h_k) / s_y + 1)
+        >>> w_o = int((w_i + 2 * w_p - w_k) / s_x + 1)
+        >>> y.shape == (n, c_o, h_o, w_o)
+        True
```
Comment (delta2323) [inline chainer/functions/connection/convolution_2d.py]: Please add the example for `cover_all=False` case.

**Comment by keisuke-umezawa** [comment]
I fixed the documents. Please merge after the test.

**Comment by delta2323** [comment]
LGTM

**Pull request merged**